### ISSUE TITLE
Instances stuck in "stopping" state with exception - java.lang.IllegalStateException: Attempt to cancel when process is still transitioning

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#1714]
Build from master - Aug1 

Instances are stuck in "stopping" state forever . "java.lang.IllegalStateException: Attempt to cancel when process is still transitioning" exception seen in logs:

The services that these instances belong to are in "Removed" state.

mysql> select id,state,name from instance where id = 2882;
+------+----------+------------------------------------+
| id   | state    | name                               |
+------+----------+------------------------------------+
| 2882 | stopping | test427569_test133814_test539859_1 |
+------+----------+---------------

mysql> select service_id, instance_id from service_expose_map where instance_id=2882;
+------------+-------------+
| service_id | instance_id |
+------------+-------------+
|        630 |        2882 |
+------------+-------------+
1 row in set (0.01 sec)

mysql> select name,id,state from service where id = 630;
+------------+-----+---------+
| name       | id  | state   |
+------------+-----+---------+
| test133814 | 630 | removed |
+------------+-----+---------+
1 row in set (0.00 sec)

mysql> 

```
2015-07-31 16:00:12,251 ERROR [:] [] [] [] [utorService-193] [i.c.p.e.e.i.ProcessEventListenerImpl] Unknown exception running process [instance.stop:75510] on [2782] java.lang.IllegalStateExceptio
n: Attempt to cancel when process is still transitioning
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.runDelegateLoop(DefaultProcessInstanceImpl.java:194) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.executeWithProcessInstanceLock(DefaultProcessInstanceImpl.java:161) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$1.doWithLock(DefaultProcessInstanceImpl.java:110) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$1.doWithLock(DefaultProcessInstanceImpl.java:107) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl$3.doWithLock(AbstractLockManagerImpl.java:40) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.LockManagerImpl.doLock(LockManagerImpl.java:33) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:13) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:37) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.execute(DefaultProcessInstanceImpl.java:107) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.eventing.impl.ProcessEventListenerImpl.processExecute(ProcessEventListenerImpl.java:66) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at sun.reflect.GeneratedMethodAccessor471.invoke(Unknown Source) ~[na:na]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_79]
        at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_79]
        at io.cattle.platform.eventing.annotation.MethodInvokingListener$1.doWithLockNoResult(MethodInvokingListener.java:76) [cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:7) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:3) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl$3.doWithLock(AbstractLockManagerImpl.java:40) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.LockManagerImpl.doLock(LockManagerImpl.java:33) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:13) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:37) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.eventing.annotation.MethodInvokingListener.onEvent(MethodInvokingListener.java:72) [cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.eventing.impl.AbstractThreadPoolingEventService$2.doRun(AbstractThreadPoolingEventService.java:135) [cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.NoExceptionRunnable.runInContext(NoExceptionRunnable.java:13) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.ManagedContextRunnable$1.run(ManagedContextRunnable.java:49) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:55) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:108) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:52) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.ManagedContextRunnable.run(ManagedContextRunnable.java:46) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_79]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_79]
        at java.lang.Thread.run(Thread.java:745) [na:1.7.0_79]
```

**Comment by alena1108** [comment]
@sangeethah is setup still around? This error msg just indicates that the instance is stuck in stopping state, but doesn't tell the reason. I need to look at the logs to find out more.

**Comment by sangeethah** [comment]
Logs are not available from the setup anymore.
I will attach the logs when I see this issue again.

**Comment by alena1108** [comment]
@sangeethah sure, please reassign it back to me once more logs are available.

**Comment by willseward** [comment]
https://www.dropbox.com/s/echoy2uygy99a9z/setup.log?dl=0

I believe this may be the same error. I'm not sure if the out of memory exception at the end is related...

**Comment by willseward** [comment]
After increasing the RAM on the server, it looks like the last exception was not the root cause of the error. However, it may have cause the server to crash in an irrecoverable state.

**Comment by ibuildthecloud** [comment]
@sangeetha I believe this issue is fixed on master and will be in v0.36.  We need to open a separate issue that long running failing task consume too much memory.

**Comment by sangeethah** [comment]
@ibuildthecloud  @willseward #1954 is logged to track the issue with rancher server running out of memory because of long running failing tasks.

**Comment by alena1108** [comment]
Should be fixed in v0.36 as per @ibuildthecloud comment; @sangeethah could you please retest?

**Comment by sangeethah** [comment]
This issue is not seen when running automation runs where environment deletion gets done on every successful test execution.
